### PR TITLE
add name space for joint slider

### PR DESCRIPTION
**Pull request by lianghongzhuo**
## Proposed changes

The current joint slider has a hard-coded namespace that assumes only the hand is running.
However, when running the hand together with an arm, it is common to set a namespace for each component, (in our case it's called `/arm/***` and `/hand/***`, so the current code here cannot find correct controller as the real name is `/hand/controller_manager/list_controllers` instead of just `controller_manager/list_controllers`
I added a namespace prefix in the init function, so people who have a robot system like us can add the correct namespace prefix there to run the joint slider successfully.
This change has tested on our robot system.

## Checklist

If the task is applicable to this pull request (see applicability criteria in brackets), make sure it is completed before checking the corresponding box. Otherwise, tick the box right away. Make sure that **ALL** boxes are checked **BEFORE** the PR is merged.

- [x] Manually tested that added code works as intended (if any functional/runnable code is added).
- [x] Added automated tests (if a new class is added (Python or C++), interface of that class must be unit tested).
- [x] Tested on real hardware (if the changed or added code is used with the real hardware).

### CODE REVIEW

**Review by beatrizleon — DISMISSED** [review]

File: sr_gui_joint_slider/src/sr_gui_joint_slider/joint_slider.py [hunk 1]
```diff
@@ -66,6 +66,7 @@ class SrGuiJointSlider(Plugin):
 
     def __init__(self, context):
         super(SrGuiJointSlider, self).__init__(context)
+        self.hand_namespace = ""
```
Comment (carebare47) [inline sr_gui_joint_slider/src/sr_gui_joint_slider/joint_slider.py]: Could this be parameterised so that it can be passed in?

**Comment by lianghongzhuo** [comment]
one way is to let the user post a ros parameter that indicates the hand namespace?
in user's launch file, set:
```
<param name="shadow_hand_namespace" command="/hand/" />
```
we then read this parameter in jont_slider.py if we can't read this parameter, then set name_space to ""

File: sr_gui_joint_slider/src/sr_gui_joint_slider/joint_slider.py (hunk 1 above)
Comment (jeferal) [inline sr_gui_joint_slider/src/sr_gui_joint_slider/joint_slider.py]:
```suggestion
        self.hand_namespace = rospy.get_param("~shadow_hand_namespace", "")
```

**Comment by toliver** [comment]
I would propose a different way to achieve the goal.

The user can simply launch rqt inside their desired namespace in order to run the joint states plugin.
You can in fact launch more than one instance of rqt, so you could have one launched inside the `/hand` namespace
and then have another rqt launched without a namespace for the plugins that require being at the root, or in the `/arm/` namespace if you need that.

That looks simpler to me than having to set parameters for the plugin.
That's the default behaviour when you call the param reads or the subscribe etc without using a leading `/` in the name.
It's good practice to not use the `/` precisely because then you make the name relative to the node's namespace level
so something like:
`rosrun rqt rqt __ns="/hand"`
should do the trick.

**Comment by lianghongzhuo** [comment]
@toliver Hi, I did not find any commands that can pass the namespace like that.
Could you illustrate it in detail?
`rqt __ns="/hand"` will give the error `rqt: error: unrecognized arguments: __ns=/hand`

**Comment by jeferal** [comment]
Could you try the following?
```
rosrun rqt_gui rqt_gui __ns:="hand"
```

**Comment by lianghongzhuo** [comment]
> `rosrun rqt_gui rqt_gui __ns:="hand"`

This command works! Thank you.
Although I have to apply the patch https://github.com/shadow-robot/sr_visualization_common/pull/37 to make the code work.

**Comment by lianghongzhuo** [comment]
Since the above-mentioned command is also working, I will close this PR.

**Pull request closed without merging**